From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: deepak1556 <hop2deep@gmail.com>
Date: Wed, 20 Aug 2025 04:03:11 +0900
Subject: chore: add electron objects to WrappablePointerTag

Extends gin::WrappablePointerTag with tags needed for
electron objects that extend gin::Wrappable and gets
allocated on the cpp heap

diff --git a/gin/public/wrappable_pointer_tags.h b/gin/public/wrappable_pointer_tags.h
index c29e8554933994ff56ccea394af34e17c4e9fc2c..42512541b36ceb353483a29eca2c858b9628854b 100644
--- a/gin/public/wrappable_pointer_tags.h
+++ b/gin/public/wrappable_pointer_tags.h
@@ -76,7 +76,19 @@ enum WrappablePointerTag : uint16_t {
   kTextInputControllerBindings,  // content::TextInputControllerBindings
   kWebAXObjectProxy,             // content::WebAXObjectProxy
   kWrappedExceptionHandler,      // extensions::WrappedExceptionHandler
-  kLastPointerTag = kWrappedExceptionHandler,
+  kElectronApp,                         // electron::api::App
+  kElectronDataPipeHolder,              // electron::api::DataPipeHolder
+  kElectronDebugger,                    // electron::api::Debugger
+  kElectronEvent,                       // gin_helper::internal::Event
+  kElectronMenu,                        // electron::api::Menu
+  kElectronNetLog,                      // electron::api::NetLog
+  kElectronPowerMonitor,                // electron::api::PowerMonitor
+  kElectronPowerSaveBlocker,            // electron::api::PowerSaveBlocker
+  kElectronReplyChannel,                // gin_helper::internal::ReplyChannel
+  kElectronScreen,                      // electron::api::Screen
+  kElectronSession,                     // electron::api::Session
+  kElectronWebRequest,                  // electron::api::WebRequest
+  kLastPointerTag = kElectronWebRequest,
 };
 
 static_assert(kLastPointerTag <
